pkg/util/deletion/deletion.go

```diff
@@ -0,0 +1,58 @@
+package deletion
+
+import (
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/logql"
+	"github.com/grafana/loki/pkg/logql/log"
+	"github.com/grafana/loki/pkg/logql/syntax"
+)
+
+func SetupPipeline(req logql.SelectLogParams, p log.Pipeline) (log.Pipeline, error) {
```

Comment:
I think it could go there. I do thing this does a good job communicating what happens when it's used, though

```go
pipeline, err = deletion.SetupPipeline(req, pipeline)
if err != nil {
	return nil, err
}
```